app/lib/registries/full_topic_taxonomy_registry.rb

```diff
@@ -6,6 +6,17 @@ def [](base_path)
       taxonomy[base_path]
     end
 
+    def find_parent_by_id id
+      topic = taxonomy.find { |_, value|
+        value['content_id'] == id
+      }
+
+      title = topic.second['title']
+      path = topic.first
+
+      title.nil? ? nil : {path: path, title: title}
```

Comment:
```Find``` is a linear search and not very efficient - would it make sense to construct a second hash indexing on content_id?

Reply:
Would we cache this second hash?

Reply:
This is fixed by #1224.